fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DbFedoraToOcflObjectIndex.java

```diff
@@ -0,0 +1,117 @@
+import org.fcrepo.persistence.ocfl.api.FedoraToOcflObjectIndex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.DefaultResourceLoader;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.jdbc.datasource.init.DatabasePopulatorUtils;
+import org.springframework.jdbc.datasource.init.ResourceDatabasePopulator;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.sql.DataSource;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * Maps Fedora IDs to the OCFL IDs of the OCFL objects the Fedora resource is stored in. This implementation is backed
+ * by a relational database.
+ */
+@Component
+public class DbFedoraToOcflObjectIndex implements FedoraToOcflObjectIndex {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DbFedoraToOcflObjectIndex.class);
+
+    private static final String DDL = "sql/default-ocfl-index.sql";
+
+    private static final String MAPPING_TABLE = "ocfl_id_map";
+
+    private static final String FEDORA_ID_COLUMN = "fedora_id";
+
+    private static final String FEDORA_ROOT_ID_COLUMN = "fedora_root_id";
+
+    private static final String OCFL_ID_COLUMN = "ocfl_id";
+
+    private static final String LOOKUP_MAPPING = "SELECT * FROM " + MAPPING_TABLE +
+            " WHERE " + FEDORA_ID_COLUMN + " = :fedoraId";
+
+    private static final String INSERT_MAPPING = "INSERT INTO " + MAPPING_TABLE +
+            " (" + FEDORA_ID_COLUMN + ", " + FEDORA_ROOT_ID_COLUMN + ", " + OCFL_ID_COLUMN + ")" +
+            " VALUES (:fedoraId, :fedoraRootId, :ocflId)";
+
+    private static final String REMOVE_MAPPING = "DELETE FROM " + MAPPING_TABLE +
+            " WHERE " + FEDORA_ID_COLUMN + " = :fedoraId";
+
+    private static final String TRUNCATE_MAPPINGS = "TRUNCATE TABLE " + MAPPING_TABLE;
+
+    private final DataSource dataSource;
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+
+    public DbFedoraToOcflObjectIndex(@Autowired final DataSource dataSource) {
+        this.dataSource = dataSource;
+        this.jdbcTemplate = new NamedParameterJdbcTemplate(dataSource);
+    }
+
+    @PostConstruct
+    public void setup() {
+        LOGGER.info("Applying ddl: {}", DDL);
+        DatabasePopulatorUtils.execute(
+                new ResourceDatabasePopulator(new DefaultResourceLoader().getResource("classpath:" + DDL)),
+                dataSource);
+    }
+
+    @Override
+    public FedoraOCFLMapping getMapping(final String fedoraId) throws FedoraOCFLMappingNotFoundException {
+        try {
+            return jdbcTemplate.queryForObject(LOOKUP_MAPPING,
+                    Map.of("fedoraId", fedoraId), (rs, rowNum) -> {
+                        return new FedoraOCFLMapping(rs.getString(2), rs.getString(3));
```

Comment:
Presumably the magic numbers of "2" and "3" map to the columns "FEDORA_ROOT_ID_COLUMN" and "OCFL_ID_COLUMN"?
Can you make that more transparent... and obvious for potential future debugging/refactoring?

Reply:
Added a comment